yt_dlp/extractor/instagram.py

```diff
@@ -42,13 +43,23 @@ class InstagramBaseIE(InfoExtractor):
 
     _API_BASE_URL = 'https://i.instagram.com/api/v1'
     _LOGIN_URL = 'https://www.instagram.com/accounts/login'
-    _API_HEADERS = {
-        'X-IG-App-ID': '936619743392459',
-        'X-ASBD-ID': '198387',
-        'X-IG-WWW-Claim': '0',
-        'Origin': 'https://www.instagram.com',
-        'Accept': '*/*',
-    }
+
+    def _api_headers(self):
+        base_headers = {
+            'X-IG-App-ID': '936619743392459',
+            'X-ASBD-ID': '198387',
+            'X-IG-WWW-Claim': '0',
+            'Origin': 'https://www.instagram.com',
+            'Accept': '*/*',
+        }
+
+        headers = {}
+        for k, v in base_headers.items():
+            if k in self.get_param('http_headers'):
+                headers[k] = self.get_param('http_headers')[k]
```

Comment:
this will overwrite the `Accept` header every time

Reply:
Ah, yes, I see there's an inherent value of `'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8'` (though it does not seem to affect the response).